creator-node/src/routes/files.js

```diff
@@ -174,69 +181,69 @@ module.exports = function (app) {
     logger.info(`IPFS Standalone Request - ${CID}`)
     logger.info(`IPFS Stats - Standalone Requests: ${totalStandaloneIpfsReqs}`)
 
-    // Conditionally rehydrate from filestorage to IPFS.
-    try {
-      await rehydrateIpfsFromFsIfNecessary(
-        req,
-        CID,
-        queryResults.storagePath
-      )
-    } catch (e) {
-      // If rehydrate throws error, return 500 without attempting to stream file.
-      return sendResponse(req, res, errorResponseServerError(e.message))
-    }
-
     // If client has provided filename, set filename in header to be auto-populated in download prompt.
     if (req.query.filename) {
       res.setHeader('Content-Disposition', contentDisposition(req.query.filename))
     }
 
-    // Stream file to client.
-    try {
-      // Cat 1 byte of CID in ipfs to determine if file exists
-      // If the request takes under 500ms, stream the file from ipfs
-      // else if the request takes over 500ms, throw an error and stream the file from file system
-      await ipfsSingleByteCat(CID, req, 500)
+    if (req.query.fromFS && req.query.fromFS === 'true') {
+      logger.info(`Retrieving directly from filesystem`)
+      // Retrieves the file directly from the filesystem rather than checking IPFS first
+
+      // Fire-and-forget rehydration
+      rehydrateIpfsFromFsIfNecessary(
```

Comment:
jc, if this throws an error, will that fail this query?

Reply:
also it looks like we rehydrate in both if/else cases. should we just always rehydrate if we hit this route?

Reply:
one is awaited, the other is not, there's an important distinction -- in this case here, it will call the method and return the result immediately